src/components/atoms/InputText/InputText.tsx

```diff
@@ -3,77 +3,66 @@ import styled, { css } from 'styled-components';
 import { colors } from '../../../constants';
 import { getInputTextColor, getBorderColorByStatus } from '../../../helpers/utils';
 import { InputProps } from '../../types';
-import { useThemeContext, AppTheme } from '../../styles/Theme';
+import { useThemeContext, AppThemeProps } from '../../styles/Theme';
 
-type IconWrapperProps = {
-  startIcon?: boolean;
-  /**
-   * The weight of the rendered text.
-   * @default 'regular'
-   */
-  weight?: keyof AppTheme['typography']['weights'];
-  /**
-   * The size you want to render your text at, currently only `13px` | `15px` and `18px` supported.
-   * @default 'body'
-   */
-  size?: keyof AppTheme['typography']['text']['sizes'];
-};
+interface InputThemeProps extends AppThemeProps, InputProps {}
 
 const InputWrapper = styled.div`
   position: relative;
 `;
 
-const IconWrapper = styled.span<IconWrapperProps & { theme: AppTheme }>`
+const IconWrapper = styled.span<InputThemeProps>`
   position: absolute;
   top: 1px;
   bottom: 1px;
   display: flex;
   align-items: center;
   justify-content: center;
-  width: 48px;
-  color: ${({ theme }) => theme.input.iconColor};
-  ${({ startIcon }) =>
-    startIcon
-      ? css`
-          left: 1px;
-          border-top-left-radius: 8px;
-          border-bottom-left-radius: 8px;
-        `
-      : css`
-          right: 1px;
-          border-top-right-radius: 8px;
-          border-bottom-right-radius: 8px;
-        `}
+  width: ${({ theme }: InputThemeProps) => (theme?.input?.iconWidth ? theme.input.iconWidth : '48px')};
+  color: ${({ theme }: InputThemeProps) => theme.input.iconColor};
+${({ startIcon }) =>
+  startIcon
+    ? css`
+        left: 1px;
+        border-top-left-radius: 8px;
+        border-bottom-left-radius: 8px;
+      `
+    : css`
+        right: 1px;
+        border-top-right-radius: 8px;
+        border-bottom-right-radius: 8px;
+      `}
   background-color: ${({ startIcon, theme }) => startIcon && theme.input.iconBackgroundColor};
 `;
 
-const Input = styled.input<InputProps & { theme: AppTheme }>`
+const Input = styled.input<InputThemeProps>`
   width: 100%;
   -webkit-appearance: none;
   outline: none;
-  border-radius: ${({ theme }) => theme.input.borderRadius};
+  border-radius: ${({ theme }: InputThemeProps) => theme.input.borderRadius};
   height: 50px;
-  padding: 0 16px;
-  padding-left: ${({ startIcon }) => !!startIcon && '60px'};
-  padding-right: ${({ endIcon }) => !!endIcon && '60px'};
-  font-size: ${({ theme, fontSize = 'body' }) => theme.typography.text.sizes[fontSize]};
-  font-weight: ${({ theme, fontWeight = 'regular' }) => theme.typography.weights[fontWeight]};
+  padding: ${({ theme }: InputThemeProps) => theme.input.padding};
+  padding-left: ${({ startIcon, theme }: InputThemeProps) =>
+    startIcon && theme.input.startIcon ? theme.input.startIconPaddingLeft : null};
+  padding-right: ${({ endIcon }: InputThemeProps) => !!endIcon && '60px'};
```

Comment:
Try to keep it consistent:

Something like this for padding:

padding-left: ${({ startIcon, theme }) => (startIcon ? theme.input.startIconPaddingLeft : theme.input.padding)};
  padding-right: ${({ endIcon, theme }) => (endIcon ? '60px' : theme.input.padding)};

Reply:
will touch this when we do dropdown as that impacts the padding right/end icon so this will ultimately end up changing, but i agree!